ppoprf/src/lib.rs

```diff
@@ -15,8 +15,30 @@
 pub mod ggm;
 pub mod ppoprf;
 
+use derive_more::{Display, Error};
+
+#[derive(Debug, Error, Display, PartialEq)]
+pub enum PPRFError {
+    #[display(
+        fmt = "Tag index is out of bounds, indicated index {} is above length of {}",
+        index,
+        tag_size
+    )]
+    BadTagIndex { index: usize, tag_size: usize },
+    #[display(fmt = "No prefix found")]
+    NoPrefixFound,
```

Comment:
The `NoPrefixFound` error is more internal than the rest, and so I'm not sure that we will really want to surface this to the user if it occurs (since it won't be very informative). Maybe we could just have a generic error message that says "Something went wrong"?